create.ps1

```diff
@@ -1,258 +1,449 @@
+                            #endregion Set ManagerId
+
+                            $outputContext.AuditLogs.Add([PSCustomObject]@{
+                                    # Action  = "" # Optional
+                                    Message = "Set manager for created account with AccountReference: $($outputContext.AccountReference | ConvertTo-Json). New value: $($currentKPNLisaManagerAccountId | ConvertTo-Json)."
+                                    IsError = $false
+                                })
+                        }
+                        else {
+                            Write-Warning "DryRun: Would set manager for created account. New value: $($currentKPNLisaManagerAccountId | ConvertTo-Json)"
+                        }
+                        #endregion Set Manager
+
+                        break
+                    }
+
+                    "AccountReferenceEmpty" {
+                        #region No account found
+                        $actionMessage = "setting manager for created account"
+
+                        $outputContext.AuditLogs.Add([PSCustomObject]@{
+                                # Action  = "" # Optional
+                                Message = "Skipped setting manager for created account with AccountReference: $($outputContext.AccountReference | ConvertTo-Json). Reason: Manager AccountReference is empty."
+                                IsError = $false
+                            })
+                        #endregion No account found
+        
+                        break
+                    }
+                }
             }
+            #endregion Manager
+
+            break
+        }
+
+        "Correlate" {
+            #region Correlate account
+            $actionMessage = "correlating to account"
+
+            $outputContext.AccountReference = "$($correlatedAccount.id)"
+            $outputContext.Data = $correlatedAccount.PsObject.Copy()
 
             $outputContext.AuditLogs.Add([PSCustomObject]@{
-                    Action  = "CreateAccount" # Optionally specify a different action for this audit log
-                    Message = "Added Manager $($personContext.Manager.displayName) to '$($personContext.Person.DisplayName)'"
-                    IsError = $False
+                    Action  = "CorrelateAccount" # Optionally specify a different action for this audit log
+                    Message = "Correlated to account with AccountReference: $($outputContext.AccountReference | ConvertTo-Json) on [$($correlationField)] = [$($correlationValue)]."
+                    IsError = $false
                 })
+
+            $outputContext.AccountCorrelated = $true
+            #endregion Correlate account
+
+            break
         }
 
-        $outputContext.Success = $True
+        "MultipleFound" {
+            #region Multiple accounts found
+            $actionMessage = "correlating to account"
```

Comment:
Mulitple accounts found.

Reply:
Dat is het bericht ja, maar de actionMessage is er om aan te geven bij welke actie het mis gaat.
En het gaat mis bij het correleren, omdat er meer accounts gevonden zijn, kan helloid niet kiezen welke.
Dus de actionMessage klopt.